src/00/02-introspect/proposal.adoc

```diff
@@ -0,0 +1,879 @@
+EcoStd. footnote:EcoStd[Ecosystem Standards (https://ecostd.github.io/)]
+
+Vendor:: Any other capability, i.e. other than `std`, is available for vendors
+to use as extensions outside the EcoStd. footnote:EcoStd[]
+
+There was a question on "Why not alow 0-9 in the name?". Considering this brings
+up the question as to the utility of having numbers in the name. An
+obvious use case is to add versioning to the name, for example `std2`. That is a
+case we want to avoid. As it avoids using the version numbers themselves which
+subverts the spirit of the introspection. Another use case is to cover vendor
+specific names for tools that use a number in their names, for example `b2`,
+`build2`. Because that is a currently existing use case, and that forcing such
+applications that want custom capabilities to create alternate names has
+various drawbacks, yes, we should accept numbers in the names.
+
+=== Version Specification
+
+When indicating the version, or versions, to the target tool or the consumer the
+version information is specified in two possible forms: a single version, or a
+single version range.
+
+==== Semantic Versioning
+
+We use the base (pre-release and build labels are not allowed) specification of
+Semantic Versioning 2.0.0
+footnote:SemVer[Semantic Versioning 2.0.0 (https://semver.org/spec/v2.0.0.html)]
+to define the syntax and semantics of compatibility.
+
+--
+We define a tool (producer or consumer) to be _backward compatible_, for
+semantic versioning, with another tool (consumer or producer) when the
+consumer that implements an older version of the API can operate, with the same
+semantics, when interacting with a producer that implements a newer version of
+the API, and vice versa.
+
+For example: If a producer generates JSON structured data. In a newer,
+compatible, version if may decided to introduce a new field. If such a field
+can be ignored by the consumer such that ignoring it does not change the
+operational semantics of the consumer the API would be considered
+_backward compatible_. And hence could be indicated with a MINOR or PATCH
+version difference per semantic versioning.
+
+The specifics of how the API behaves to achieve _backward compatible_ changes
+is up to the individual specification of the capabilities. As the ability to be
+_backward compatible_ varies with the specifics of many factors, like tool
+options, data formats, and so on.
+--
+
+==== Single Version
+
+A single version in this proposal is composed of a one to three dotted whole
+numbers. The numbers are expected to be strictly increasing.
+Following SemVer footnote:SemVer[] a change to the MAJOR version indicates a
+backward incompatible change. And changes to the MINOR and PATCH versions
+indicate backward compatible changes.
+The format for the version must match the regular expression: footnote:Regex[]
+
+[source,plain_text]
+----
+^[0-9]+([.][0-9]+){0,2}$
+----
+
+==== Version Range
```

Comment:
We should adopt pre-existing syntax and specifications from other languages for version constraints[^1][^2]. We don't need to invent a new one. 

[^1]: https://python-poetry.org/docs/dependency-specification
[^2]: https://github.com/npm/node-semver